src/components/Api.js

```diff
@@ -0,0 +1,129 @@
+class Api {
+  constructor(options) {
+    this.options = options;
+  }
+
+  // Common method for handling fetch requests
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return res.json().then((data) => Promise.reject(data));
+  }
+
+  // Common method for handling fetch errors
+  _handleError(err) {
+    console.error(`Error: ${err.message}\nURL: ${err.url}`);
+  }
+
+  // Get user information
+  getUserInfo() {
+    return fetch(`${this.options.baseUrl}/users/me`, {
+      headers: {
+        authorization: this.options.headers.authorization,
+      },
+    })
+      .then(this._handleResponse)
+      .catch(this._handleError);
+  }
+
+  // Update user profile
+  updateUserProfile(data) {
+    return fetch(`${this.options.baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        authorization: this.options.headers.authorization,
+        "Content-Type": "application/json",
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://disk.yandex.ru/i/J93dbJp7eypXKQ . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)